test/src/folding/foldable_block_test.dart

```diff
@@ -0,0 +1,148 @@
+import 'package:flutter_code_editor/src/folding/foldable_block.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'parsers/test_executor.dart';
+
+void main() {
+  group('Foldable block test', () {
+    test('Join intersected blocks in empty list will not throw exception', () {
+      const blocks = <FoldableBlock>[];
+
+      final actual = blocks.deepCopy()..joinIntersecting();
+
+      expect(actual, blocks);
+    });
+
+    test('Join blocks in list with one block will not throw exception', () {
+      const blocks = [
+        FoldableBlock(startLine: 0, endLine: 1, type: FBT.braces),
+      ];
+
+      final actual = blocks.deepCopy()..joinIntersecting();
+
+      expect(actual, blocks);
+    });
+
+    test('Join intersecting blocks in list with two intersecting blocks', () {
+      const blocks = [
+        FoldableBlock(startLine: 0, endLine: 1, type: FBT.braces),
+        FoldableBlock(startLine: 1, endLine: 2, type: FBT.braces),
+      ];
+      const matcher = [
```

Comment:
`Matcher` is a Dart class that does the comparison. This may be called 'expected' or alike.